lib/screens/home/home_screen.dart

```diff
@@ -1,18 +1,284 @@
 
-class HomeScreen extends StatelessWidget {
+class HomeScreen extends ConsumerStatefulWidget {
   const HomeScreen({super.key});
 
+  @override
+  ConsumerState<ConsumerStatefulWidget> createState() => HomeScreenState();
+}
+
+class HomeScreenState extends ConsumerState<HomeScreen> {
+  TextTheme get _textTheme => Theme.of(context).textTheme;
+
+  AppLocalizations get _localizations => AppLocalizations.of(context)!;
+  int _selectedPage = 0;
+
+  final _shouldShowShimmerProvider = StreamProvider.autoDispose<bool>(
+      (ref) => ref.watch(homeViewModelProvider.notifier).shouldShowShimmer);
+  final _currentDateStreamProvider = StreamProvider.autoDispose(
+      (ref) => ref.watch(homeViewModelProvider.notifier).currentDate);
+  final _surveysStreamProvider = StreamProvider.autoDispose(
+      (ref) => ref.watch(homeViewModelProvider.notifier).surveys);
+  final _profileStreamProvider = StreamProvider.autoDispose(
+      (ref) => ref.watch(homeViewModelProvider.notifier).profile);
+  final _isLoadMoreStreamProvider = StreamProvider.autoDispose(
+      (ref) => ref.watch(homeViewModelProvider.notifier).isLoadMore);
+  final _isErrorStreamProvider = StreamProvider.autoDispose(
+      (ref) => ref.watch(homeViewModelProvider.notifier).isError);
+
+  late PageController _pageController;
+
+  @override
+  void initState() {
+    _pageController = PageController(initialPage: _selectedPage);
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    _pageController.dispose();
+    super.dispose();
+  }
+
   @override
   Widget build(BuildContext context) {
-    // TODO Add home screen UI in part 2
-    return const Scaffold(
+    ref.listen<AsyncValue<String>>(_isErrorStreamProvider, (previous, next) {
+      final error = next.value ?? '';
+      if (error.isNotEmpty) {
+        WidgetsBinding.instance
+            .addPostFrameCallback((_) => createSnackBar(error));
+        ref.read(homeViewModelProvider.notifier).clearError();
+      }
+    });
+
+    return Scaffold(
+      body: Consumer(builder: (context, widgetRef, child) {
+        var isShimmerLoading =
+            widgetRef.watch(_shouldShowShimmerProvider).value ?? false;
+        final surveys = widgetRef.watch(_surveysStreamProvider).value ?? [];
+        final imageUrl =
+            widgetRef.watch(_profileStreamProvider).value?.imageUrl ?? '';
+        final isLoadMore =
+            widgetRef.watch(_isLoadMoreStreamProvider).value ?? false;
+        final currentDate =
+            widgetRef.watch(_currentDateStreamProvider).value ?? '';
+
+        if (isShimmerLoading) {
+          return const HomeShimmerLoading();
+        } else {
+          return _buildHomeScreenContent(
+              surveys: surveys,
+              imageUrl: imageUrl,
+              isLoadMore: isLoadMore,
+              currentDate: currentDate,
+              onLoadMore: () {
+                widgetRef.read(homeViewModelProvider.notifier).getSurveyList();
+              });
```

Comment:
```suggestion
              },
          );
```
Please don't forget to put trailing comma at the end 😁